src/Microsoft.AspNet.Mvc.Core/DefaultControllerActionArgumentBinder.cs

```diff
@@ -118,19 +117,18 @@ private static void CallPropertyAddRange<TElement>(object target, object source)
 
         private void ActivateProperties(object controller, Type containerType, Dictionary<string, object> properties)
         {
-            var propertyHelpers = PropertyHelper.GetProperties(controller);
+            var controllerMetadata = _modelMetadataProvider.GetMetadataForType(containerType);
```

Comment:
I'll revert a bunch of this. But I'm still curious about the side discussion around `new`...

> It should be totally viable to use `new` in a controller to hide services used by the `Controller` class itself.

Services? A user can derive their controller classes from whatever they like and their base could use other model binders for properties.

But my push back is the same if focused on `Controller` in particular. In the following code, bad things could happen if `IMyMetadataProvider` and `IModelMetadataProvider` are different types, perhaps because only this controller needs or wants the extra `IMyMetadataProvider` behaviours.
``` c#
    public class MyController : Controller
    {
        [FromServices]
        public new IMyMetadataProvider MetadataProvider
        {
            get { return (IMyMetadataProvider)base.MetadataProvider; }
            set { base.MetadataProvider = value; }
        }
    }
```

Should I open an issue for further discussion of this point?